### PR TITLE
feat: init Commlint✨ 

### PR DESCRIPTION
**Pull request by Nishitbaria**
## Related Issue

Closes: #599 

## Description of Changes

install the Commit lint into project
install the  husky 

## Checklist:

- [ ] My code adheres to the established style guidelines of this project.
- [x] I have conducted a self-review of my code.
- [ ] I have included comments in areas that may be difficult to understand.
- [ ] I have made corresponding updates to the project documentation.
- [x] My changes have not introduced any new warnings.

## Screenshots

Screenshot (20)

Please provide any necessary screenshots to illustrate the changes made.

### CODE REVIEW

**Comment by Nishitbaria** [comment]
Hey @anmode  Please Review my pr

**Comment by Nishitbaria** [comment]
@anmode  Comment Lint is used for Conventional Commits whereas Prettier is used to format the code both are different things, man

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -117,6 +117,57 @@ Effective communication is essential for maintaining a collaborative and inclusi
...
+- `auth feature update`
+- `chore: fixed some stuff`
+
+### Commit Example with Commitlint:
+
+```bash
+git commit -m "feat(auth): Implement user signup process (#789)"
+```
+
+Remember to run `commitlint` before pushing your changes to ensure your commit messages meet the guidelines.
```
Comment (anmode) [inline CONTRIBUTING.md:167]: yeh run krna pdega? I mean aisa ni h if contributor try to push and push n ho as commit message do not meet requirements?

Comment (Nishitbaria) [reply CONTRIBUTING.md:167]: Na ye automatic ho jayega

**Pull request merged**